src/main/QueryServer.cpp

```diff
@@ -184,10 +210,218 @@ QueryServer::getLedgerEntryRaw(std::string const& params,
+
+    for (auto const& key : keys)
+    {
+        LedgerKey k;
+        fromOpaqueBase64(k, key);
+        if (k.type() == TTL)
+        {
+            retStr = "TTL keys are not allowed\n";
+            return false;
+        }
+
+        auto [_, inserted] = keysToSearch.emplace(k);
+        if (!inserted)
+        {
+            retStr = "Duplicate keys\n";
+            return false;
+        }
+
+        inputOrderedKeys.push_back(k);
+    }
+
+    mBucketSnapshotManager.maybeCopyLiveAndHotArchiveSnapshots(liveBl,
+                                                               hotArchiveBl);
+
+    std::vector<LedgerEntry> liveEntries;
+    std::vector<HotArchiveBucketEntry> archivedEntries;
+    uint32_t ledgerSeq =
+        snapshotLedger ? *snapshotLedger : liveBl->getLedgerSeq();
+    root["ledger"] = ledgerSeq;
+
+    auto liveEntriesOp = liveBl->loadKeysFromLedger(keysToSearch, ledgerSeq);
+
+    // Return 404 if ledgerSeq not found
+    if (!liveEntriesOp)
+    {
+        retStr = "Ledger not found\n";
+        return false;
+    }
+
+    liveEntries = std::move(*liveEntriesOp);
+    liveEntriesOp->clear();
+
+    // Remove keys found in live bucketList from subsequent searches
+    for (auto const& le : liveEntries)
+    {
+        keysToSearch.erase(LedgerEntryKey(le));
+    }
+
+    LedgerKeySet hotArchiveKeysToSearch;
+    for (auto const& lk : keysToSearch)
+    {
+        if (isSorobanEntry(lk))
+        {
+            hotArchiveKeysToSearch.emplace(lk);
+        }
+    }
+
+    // Only query archive for soroban keys we didn't find in the live bucketList
+    if (!hotArchiveKeysToSearch.empty())
+    {
+        auto archivedEntriesOp =
+            hotArchiveBl->loadKeysFromLedger(hotArchiveKeysToSearch, ledgerSeq);
+        if (!archivedEntriesOp)
+        {
+            retStr = "Ledger not found\n";
+            return false;
+        }
+        archivedEntries = std::move(*archivedEntriesOp);
+    }
+
+    // Collect TTL keys for Soroban entries in the live BucketList
+    LedgerKeySet ttlKeys;
+    for (auto const& le : liveEntries)
+    {
+        if (isSorobanEntry(le.data))
+        {
+            ttlKeys.emplace(getTTLKey(le));
+        }
+    }
+
+    std::vector<LedgerEntry> ttlEntries;
+    if (!ttlKeys.empty())
+    {
+        // We haven't updated the live snapshot so we know the have a snapshot
+        // available for ledgerSeq
+        ttlEntries =
+            std::move(liveBl->loadKeysFromLedger(ttlKeys, ledgerSeq).value());
+    }
+
+    std::unordered_map<LedgerKey, LedgerEntry> ttlMap;
+    for (auto const& ttlEntry : ttlEntries)
+    {
+        ttlMap.emplace(LedgerEntryKey(ttlEntry), ttlEntry);
+    }
+
+    // Store key -> formatted response
+    std::unordered_map<LedgerKey, Json::Value> responseEntries;
+
+    for (auto const& le : liveEntries)
+    {
+        LedgerKey lk = LedgerEntryKey(le);
+        Json::Value entry;
+        entry["e"] = toOpaqueBase64(le);
```

Comment:
I don't feel too strongly, but one-char field names seems pretty unreadable (even if the intention is that only code will read it - the client code is going to be pretty unreadable as well).

Reply:
+1. Plus, the mixing of two values inside the `e` value is also not intuitive. The docs say it contains a LedgerEntry or a LedgerKey.